yax/modules/identify_informative_hits.py

```diff
@@ -0,0 +1,217 @@
+                                                  hit_tax_ids,
+                                                  total_informative_hits)
+
+    hit_data.inclusion_tree = taxidtool.build_tree(hit_data.tax_id_data,
+                                                   None,
+                                                   informative_tax_ids,
+                                                   [])
+
+    hit_data.gis_to_taxids = taxidtool.build_gis_to_taxids(hit_data.
+                                                           inclusion_tree)
+
+    return hit_data
+
+
+def get_informative_tax_ids(min_hits, min_relative_hits, hit_tax_ids,
+                            total_informative_hits):
+    """Based on min_hit thresholds taxids are determined to be informative
+
+    The most stringent (highest) of min_hits and min_relative_hits is used as
+    a threshold that each tax id must meet or beat in order to determined
+    informative.
+
+    Parameters
+    ----------
+    min_hits : Int
+        absolute number of hits a taxid must exhibit to be determined
+        informative
+    min_relative_hits : Int
+        number of hits relative to total number of hits reported that a
+        taxid must exhibit to be determined informative
+    hit_tax_ids : dict
+        tax_id key to number_of_hits value representing the number of hits each
+        tax_id was observed to have in the alignmnet file
+
+    Returns
+    -------
+        list
+            tax_ids that have meet or beat minimum hit thresholds
+
+    """
+    hit_threshold = max(min_hits,
+                        (total_informative_hits/100*min_relative_hits))
+
+    informative_tax_ids = []
+    for tax_id, num_hits in hit_tax_ids.items():
+        if num_hits >= hit_threshold:
+            informative_tax_ids.append(tax_id)
+
+    return informative_tax_ids
+
+
+def get_hit_taxa(hit_taxa_lines):
+    """Builds dictionary from read_ids and the tax_ids it hit
+
+    A set of all tax_ids it hit is constructed for each read_id.
+
+    parameters
+    ----------
+        hit_taxa_lines : list
+            all lines from mg_wrapper output generated in an alignment module
+
+    Returns
+    -------
+        dict
+            read_id key to list of tax_ids value where the read was observed
+            to hit the tax_id in an alignment step
+
+    """
+    hit_taxa = {}
+
+    for line in hit_taxa_lines:
+        this_line = line.rstrip().split(":")
+        this_read_id = this_line[1]
+        these_hits = set()
+        for mm_distance in this_line[1:]:
+            for tax_id_hit in mm_distance:
+                these_hits.add(tax_id_hit)
+
+        hit_taxa[this_read_id] = list(these_hits)
+
+    return hit_taxa
+
+
+def get_hit_taxids(hit_data, hit_taxa, lca_dist):
+    """Counts the numer of times each tax_id was hit in alignment
+
+    For each tax_id the number of reads that informatively aligned to it are
+    counted.
+
+    Parameters
+    ----------
+        hit_data : object
```

Comment:
right, so this is the artifact I am currently building, rather than give it a name in the scope of the module I have assigned it to an attribute of the object. It is indeed incomplete.